packages/ui/src/components/Card.tsx

```diff
@@ -33,23 +26,17 @@ import {
 } from "./hooks/useTransitioningValue";
 import CardField, { clozeBlankSentinel } from "./PromptFieldRenderer";
 
-function getQAPromptContents<PT extends QAPromptTask | ApplicationPromptTask>(
-  reviewItem: ReviewAreaItem<PT>,
-): QAPromptContents {
-  switch (reviewItem.prompt.promptType) {
-    case qaPromptType:
-      return reviewItem.prompt as QAPrompt;
-    case applicationPromptType:
-      const applicationReviewItem =
-        reviewItem as ReviewAreaItem<ApplicationPromptTask>;
-      const taskParameters = applicationReviewItem.taskParameters as {
-        variantIndex: number;
-      };
-      const applicationPrompt =
-        applicationReviewItem.prompt as ApplicationPrompt;
-
-      return applicationPrompt.variants[taskParameters.variantIndex];
-    case clozePromptType:
+function getQAPromptContents<TC extends QATaskContent | ClozeTaskContent>(
+  reviewItem: ReviewAreaItem<TC>,
+): QATaskContent {
+  switch (reviewItem.spec.content.type) {
+    case TaskContentType.QA:
+      // This cast should not be needed... My IDE does not throw any error
+      // without the cast, but TSC throws:
+      // `Type 'TC' is not assignable to type 'QATaskContent'`.
+      // casting resolves the type issue...
+      return reviewItem.spec.content as QATaskContent;
```

Comment:
Very weird error... Could this be due to my IDE and TSC using different versions of typescript? Not sure how to justify this error otherwise.

Reply:
Ah, I think this is because TS can't do this kind of type narrowing inference across multiple projections. To put it another way, this should work:

```
const {content} = reviewItem.spec;
switch (content.type) {
  case TaskContentType.QA:
    return content;
```

Reply:
Darn, I was hoping this would work. Looks, like the issue is still persisting 😕

<img width="939" alt="Screen Shot 2021-08-06 at 12 00 05 PM" src="https://user-images.githubusercontent.com/8084674/128559356-c6d4301d-c85f-496c-9e8c-ae273d0bf123.png">